deploy/runtools/firesim_topology_with_passes.py

```diff
@@ -612,56 +617,59 @@ def loop_logger(instancestates: Dict[str, Any], terminateoncompletion: bool) ->
             def get_jobs_completed_local_info():
                 # this is a list of jobs completed, since any completed job will have
                 # a directory within this directory.
-                jobscompleted = os.listdir(self.workload.job_monitoring_dir)
-                rootLogger.debug("Monitoring dir jobs completed: " + str(jobscompleted))
-                return jobscompleted
-
-            jobscompleted = get_jobs_completed_local_info()
-
-
-            # this job on the instance should return all the state about the instance
-            # e.g.:
-            # if an instance has been terminated (really - is termination
-            # requested and no jobs are left, then we will have implicitly
-            # terminated
-            teardown = False
-            instancestates = execute(monitor_jobs_wrapper, self.run_farm,
-                                    jobscompleted, teardown,
-                                    self.terminateoncompletion,
-                                    self.workload.job_results_dir,
-                                    hosts=all_run_farm_ips)
+                monitored_jobs_completed = os.listdir(self.workload.job_monitoring_dir)
+                rootLogger.debug(f"Monitoring dir jobs completed: {monitored_jobs_completed}")
+                return monitored_jobs_completed
+
+            # return all the state about the instance (potentially copy back results and/or terminate)
+            is_final_run = False
```

Comment:
The core difference here is that there is now a variable to track whether this execution of `monitor_jobs_wrapper` is the final run or not.